tests/test_main.py

```diff
@@ -1 +1,26 @@
-# write your code here
+from app.main import cryptocurrency_action
+from unittest import mock
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_buy_more_cryptocurrency(mocked_get_exchange_rate):
+    mocked_get_exchange_rate.return_value = 10.2
+    current_rate = 2.3
+    assert cryptocurrency_action(current_rate) ==\
+           "Buy more cryptocurrency"
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_sell_all_cryptocurrency(mocked_get_exchange_rate):
+    mocked_get_exchange_rate.return_value = 4.1
+    current_rate = 5.1
+    assert cryptocurrency_action(current_rate) ==\
+           "Sell all your cryptocurrency"
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_do_nothing(mocked_get_exchange_rate):
+    mocked_get_exchange_rate.return_value = 8
+    current_rate = 8
+    assert cryptocurrency_action(current_rate) ==\
+           "Do nothing"
```

Comment:
Add boundary conditions, you don't check when get_exchange_rate_prediction return value equals exactly 1.05 or 0.95 * value passed in cryptocurrency_action.